ipaclient/plugins/timerule.py

```diff
@@ -0,0 +1,74 @@
+# Authors:
+#   Stanislav Laznicka <slaznick@redhat.com>
+#
+# Copyright (C) 2016  Red Hat
+# see file 'COPYING' for use and warranty information
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import six
+
+from ipaclient.frontend import MethodOverride
+from ipalib import File
+from ipalib import errors
+from ipalib.plugable import Registry
+from ipalib import _
+
+if six.PY3:
+    unicode = str
+
+register = Registry()
+
+
+def get_accesstime(options):
+    if not('icalfile' in options or 'accesstime' in options):
+        raise errors.RequirementError(name='accesstime')
+    if 'icalfile' in options:
+        if 'accesstime' in options:
+            raise errors.OverlapError(names=['time', 'icalfile'])
+        options['accesstime'] = \
+            unicode(options['icalfile'].encode('unicode-escape'))
```

Comment:
options['accesstime'] = unicode(
    options['icalfile'].encode('unicode-escape'))

You dont neeed \